kernel/cClosure.mli

```diff
@@ -45,8 +45,10 @@ type fterm =
   | FCaseT of case_info * UVars.Instance.t * constr array * case_return * fconstr * case_branch array * usubs (* predicate and branches are closures *)
   | FCaseInvert of case_info * UVars.Instance.t * constr array * case_return * finvert * fconstr * case_branch array * usubs
   | FLambda of int * (Name.t binder_annot * constr) list * constr * usubs
+  | FLambdaF of Name.t binder_annot * fconstr * fconstr
```

Comment:
FTR, this really looks like a hack, and this makes the conversion algorithm somewhat more brittle. We should discuss offline what you actually want to achieve.